### PR TITLE
Amqp source receive message

### PR DESCRIPTION
**Pull request by jcortes**
# Action
- [x] Send message

# Source
- [x] Receive message

Implementation was tested using [Solace](https://www.solace.dev/) platform that supports AMQP 1.0 protocol

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/amqp/actions/send-message/send-message.mjs [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import amqp from "../../amqp.app.mjs";
+
+export default {
+  key: "amqp-send-message",
+  name: "Send a Message",
+  description: "Send a new message to an AMQP 1.0 queue",
```
Comment (vellames-turing) [inline components/amqp/actions/send-message/send-message.mjs:6]: Could you please add a reference to `AMQP`?

File: components/amqp/actions/send-message/send-message.mjs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      const delivery = await this.amqp.sendMessage({
+        sender,
+        body,
+        message_id: messageId,
+      });
+
+      $.export("$summary", `Sucessfully sent message with Delivery ID ${delivery.id}`);
+
+    } catch (error) {
+      console.log("Error sending message", JSON.stringify(error.innerError));
```
Comment (vellames-turing) [inline components/amqp/actions/send-message/send-message.mjs:74]: We don't wanna throw the error? Looks like if some error occurs we will just console. Maybe we could console and `throw error`?

File: components/amqp/sources/receive-message/receive-message.mjs [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import amqp from "../../amqp.app.mjs";
+
+export default {
+  key: "amqp-receive-message",
+  name: "New Message",
+  description: "Emit new event for each new message in an AMQP 1.0 queue",
```
Comment (vellames-turing) [inline components/amqp/sources/receive-message/receive-message.mjs:6]: Could you please add some ref to `AMQP`, please?

File: components/amqp/sources/receive-message/receive-message.mjs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      const id = Date.parse(timestamp);
+
+      this.$emit(message, {
+        id,
+        ts: id,
+        summary: `New Message with ID ${message.message_id}`,
+      });
+
+    } catch (error) {
+      console.log("Error receiving message", JSON.stringify(error));
```
Comment (vellames-turing) [inline components/amqp/sources/receive-message/receive-message.mjs:76]: We don't wanna throw the error? Looks like if some error occurs we will just console. Maybe we could console and `throw error`?

**Review by vellames-turing — CHANGES_REQUESTED** [review]
Hi there, just some minors

File: components/amqp/sources/receive-message/receive-message.mjs [hunk 5]
```diff
@@ -73,7 +73,8 @@ export default {
       });
 
     } catch (error) {
-      console.log("Error receiving message", JSON.stringify(error));
+      console.log("Error receiving message", JSON.stringify(error.innerError));
+      throw "Error receiving message from queue";
```
Comment (vellames-turing) [inline components/amqp/sources/receive-message/receive-message.mjs:77]: I am wondering if we can use `throw error`

**Review by vellames-turing — APPROVED** [review]
Tks

**Comment by vunguyenhung** [comment]
Hello @jcortes, I have tested this PR and there're some test cases failed or needed improvement. 

Please check the test report below for more information
[AMQP_2945.pdf](https://github.com/PipedreamHQ/pipedream/files/8815076/AMQP_2945.pdf)

**Review by feyzullah — APPROVED** [review]
LGTM

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[AMQP_2945_2152.pdf](https://github.com/PipedreamHQ/pipedream/files/8845295/AMQP_2945_2152.pdf)

**Pull request merged**